### PR TITLE
Refactor main.go

### PR DESCRIPTION
**Pull request by raghavendra-talur**
This PR refactors the main.go file. This is to enable ease of use. There are two main motivations for this:
1. The integration tests will reuse some of this code in a future PR.
2. ComponentConfig of the controller-runtime is deprecated and we need to reimplement the config option reading code for Ramen Config. This is hard to implement with the current state of the code.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -94,6 +104,7 @@ func newManager() (ctrl.Manager, *ramendrv1alpha1.RamenConfig, error) {
 		utilruntime.Must(gppv1.AddToScheme(scheme))
 		utilruntime.Must(argocdv1alpha1hack.AddToScheme(scheme))
 		utilruntime.Must(clrapiv1beta1.AddToScheme(scheme))
+		utilruntime.Must(recipe.AddToScheme(scheme))
```
Comment (ShyamsundarR) [inline main.go:107]: recipe is processed by VRG, so this addition should be in the else part?

File: main.go [hunk 2]
```diff
@@ -208,7 +227,20 @@ func setupReconcilersHub(mgr ctrl.Manager) {
...
+	ctrl.SetLogger(zap.New(zap.UseFlagOptions(logOpts)))
+
+	ctrlOptions, ramenConfig := buildOptions()
+
+	if err := configureController(ramenConfig); err != nil {
+		setupLog.Error(err, "unable to configure controller")
+		os.Exit(1)
+	}
+
+	mgr, err := newManager(ctrlOptions)
```
Comment (ShyamsundarR) [inline main.go:243]: We can possibly drop the `newManager` function and invoke `ctrl.NewManager` directly here.

File: main.go (hunk 1 above)
Comment (ShyamsundarR) [reply main.go:107]: Ah ok got hidden in the view, we are fine.

File: main.go (hunk 2 above)
Comment (ShyamsundarR) [reply main.go:243]: Ack no worries.

**Review by ShyamsundarR — APPROVED** [review]

**Review by BenamarMk — APPROVED** [review]

**Pull request merged**